price/pricing/prometheus.go

```diff
@@ -0,0 +1,46 @@
+package pricing
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+
+	"github.com/prometheus/client_golang/api"
+	promV1 "github.com/prometheus/client_golang/api/prometheus/v1"
+)
+
+// NewPromClient creates new Prometheus API client.
+func NewPromClient(address, user, password string) promV1.API {
+	c, err := api.NewClient(api.Config{
+		Address:      address,
+		RoundTripper: api.DefaultRoundTripper,
+	})
+	if err != nil {
+		panic(err)
```

Comment:
Why panic? You could return an error and allow the user the handle it however is needed 🤷🏻  Especially if you take in to account the change to `buildLoadPricingProvider` func.

Reply:
i blame copy pasta